README.md

```diff
@@ -1,6 +1,96 @@
 # example-django
+
 An example repository to demonstrate Django support in Pants.
 
-See [pantsbuild.org](https://www.pantsbuild.org/) for much more detailed documentation.
+The demo consists of a set of microservices that implement an elaborate "Hello, World" site.
+This repo shows how Pants can support multiple Django services in a single repo, with
+minimal copypasta.
+
+The layout in this repos enforces a strict distinction between Django apps (reusable units of
+functionality such as models and related views), vs. services ("projects" in Django parlance) -
+deployable services composed of multiple apps and typically attached to one or more database instances.
+
+See [pantsbuild.org](https://www.pantsbuild.org/) for much more detailed documentation about Pants.
 
 See [here](https://github.com/pantsbuild/example-python) for an example of basic Python support in Pants.
+
+## Services
+
+ There are four distinct services, defined under [`helloworld/services`](helloworld/services):
+
+- `helloworld.services.admin`: Runs the Django admin UI.
+- `helloworld.services.frontend`: Serves end user requests.
+- `helloworld.services.user`: A backend service that provides user data.
+- `helloworld.services.welcome`: A backend service that generates greetings.
+
+Each service has its own `manage.py`, `gunicorn.py`, `urls.py` and `settings.py`.
+The `settings.py` inherits most of its settings from [`helloworld/settings_base.py`](helloworld/settings_base.py)
+and adds anything specific to that service.
+
+Note that we make no argument for or against microservices vs. monolithic services as a deployment
+architecture. We *do* make an argument in favor of a monorepo over multiple repos as a codebase architecture,
+and this repo demonstrates the utility of a monorepo even if (especially if!) you deploy microservices.
+
+## Apps
+
+The services are composed of four Django apps:
+
+- `helloworld.greet`: Functionality related to selecting a greeting.
+- `helloworld.person`: Functionality related to identifying the person to greet.
+- `helloworld.translate`: Functionality related to translating greetings into various languages.
+- `helloworld.ui`: Functionality related to rendering greetings to the end user.
+
+## Databases
+
+In order to demonstrate multi-db configurations, there are two databases:
+
+- `users`: User-related data.
+- `greetings`: Greeting-related data.
+
+Each app is associated with exactly one database.
+
+Currently these are sqlite. We plan to add examples of using PostgreSQL.
+
+## Useful Pants commands
+
+To run all tests:
+
+```
+./pants test ::
+```
+
+To run formatters and linters:
+
+```
+./pants fmt lint validate ::
+```
+
+To run typechecking:
+
+```
+./pants typecheck ::
+```
+
+To build deployable gunicorn .pex files for all services:
+
+```
+./pants package ::
+```
+
+## manage.py
+
+To run management commands for a service, use that service's `manage.py`, e.g.,
+
+```
+./helloworld/service/admin/manage.py runserver
```

Comment:
https://github.com/pantsbuild/pants/issues/11619

Reply:
`./pants run` is not a good fit for `manage.py` at the moment, because of the sandboxing. For example, it makes `runserver` unable to find the sqlite files, or auto-reload on file changes.   The venv sets itself up (see ./python, I should document this).

Reply:
I don't understand the sandboxing part - `./pants run` does not sandbox?

Reply:
Maybe "sandbox" isn't the right term, since this isn't literally sandboxed like a Process. But `./pants run` runs in a tmpdir under the chroot, IIUC.  So `runserver` watches those for changes which of course never happen, and it looks for the sqlite DB files relative to those, which also is not where they are. 

Or if you run `makemigrations` it'll create the migrations files under the tmpdir, and so on.